### PR TITLE
Ajax intro

### PR DESCRIPTION
**Pull request by kmatsuleva**

### CODE REVIEW

File: ajax_intro/myScript.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                if (this.readyState == 4 && this.status == 200) {
+                myFunction(this);
+                }
+            };
+            xhttp.open("GET", "catalog.xml", true);
+            xhttp.send();
+        }
+
+
+        function myFunction(xml) {
```
Comment (mgenov) [inline ajax_intro/myScript.js]: This function renders catalog so it's better to be named: renderCatalog

File: ajax_intro/myScript.js [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
-                myFunction(this);
-                }
-            };
-            xhttp.open("GET", "catalog.xml", true);
-            xhttp.send();
+function asyncLoad(url, renderCatalog) {
+    var xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function() {
+        if (this.readyState == 4 && this.status == 200) {
+            renderCatalog(this);
```
Comment (mgenov) [inline ajax_intro/myScript.js]: if function is named asyncLoad then the renderCatalog cannot be used here as it's concrete. You can rename the param to callback or cb and then to call it with the parsed text. 

```jsx
function asyncLoad(url, callback) {
    var xhttp = new XMLHttpRequest();
    xhttp.onreadystatechange = function() {
        if (this.readyState == 4 && this.status == 200) {
            callback(this);
        }
    };
    xhttp.open("GET", url, true);
    xhttp.send();
}
```

Where the usage becomes:

```jsx

asyncLoad('catalog.xml', function(result) {
   val xmlDoc = result.responseXML
   // .. the remaining code for rendering 
})
// Note: the function is placed inline for example and it's better renderCatalog to be used directly.
// asyncLoad('catalog.xml', renderCatalog) 


asyncLoad('demo.json', function(result) {
  val user = JSON.parse(result.responseText)
  document.getElementById("demo").innerHTML = user.name + " is " + user.age + " years old";
})
```

e.g you can reuse this call for XML Catalog and for the Demo.json

Comment (mgenov) [reply ajax_intro/myScript.js]: Here is an simplified example:

```jsx
function asyncPrint(url, callback) {
  callback(url)
}

function print(text) {
  console.log('getting ' + text)
}

asyncPrint('app.json', print)
```

and in ES6: 

```jsx
const asyncPrint = (url, callback) => {
  callback(url)
}

const print = (text) => {
  console.log('getting ' + text)
}

asyncPrint('app.json', print)
```

**Comment by kmatsuleva** [comment]
Notes fixed.

**Pull request merged**